lib/web/apiserver_test.go

```diff
@@ -950,6 +950,40 @@ func (s *WebSuite) TestTerminal(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *WebSuite) TestWebsocketPingLoop(c *C) {
+	// change cluster default config for keep alive interval to be ran faster
+	clusterConfig, err := services.NewClusterConfig(services.ClusterConfigSpecV3{
+		SessionRecording:    services.RecordAtNode,
+		ProxyChecksHostKeys: services.HostKeyCheckYes,
+		KeepAliveInterval:   services.NewDuration(250 * time.Millisecond),
+		LocalAuth:           services.NewBool(true),
+	})
+	c.Assert(err, IsNil)
+
+	err = s.server.Auth().SetClusterConfig(clusterConfig)
+	c.Assert(err, IsNil)
+
+	ws, err := s.makeTerminal(s.authPack(c, "foo"))
+	c.Assert(err, IsNil)
+
+	// flush out raw event (pty texts)
+	err = s.waitForRawEvent(ws, 5*time.Second)
+	c.Assert(err, IsNil)
+
+	// next frames should be just pings sent out by ping loop
+	// b/c nothing else should be sending anything at this point
+	frame2, err := ws.NewFrameReader()
```

Comment:
nit: rename `frame2` and `frame3` to `frame`

Reply:
revised